index.d.ts

```diff
@@ -675,7 +675,7 @@ declare namespace Moleculer {
 		dependencies?: string | ServiceDependency | Array<string | ServiceDependency>;
 		metadata?: any;
 		actions?: ServiceActionsSchema;
-		mixins?: Array<Partial<ServiceSchema>>;
+		mixins?: Array<Partial<ServiceSchema>> | Array<Partial<ServiceSchema>, ThisType<Service>>;
```

Comment:
`Partial<T>` only accepts one generic value, so this is not a valid type.

I'm not really understanding the use case with `ThisType` in place.  A mixin is really just a partial service schema that gets merged together with the main service schema.  What problem is this trying to solve?

Reply:
I try remove ThisType and run

Reply:
@shawnmcknight  if not use ThisType, have use it.

![image](https://user-images.githubusercontent.com/4489558/142940165-21a13387-2035-45af-8f9c-648dcaf5d638.png)


but not work because, aren't implemented.

Reply:
![image](https://user-images.githubusercontent.com/4489558/142941329-45370497-ca43-4b7e-81f6-ca1ff58fc1d5.png)

Reply:
@devalexandre `this.DbMixin` is set to equal the return value of the `start()` method of your `DbConnection` class.  What is `start()` returning?  The error message is implying that `start()` is returning `void` which makes sense for a `start()` method generally speaking, but doesn't work for what a mixin should be.

A mixin is really just an object that represents part of a service's schema that gets merged into the mixed service's schema.  While a class method could certainly return that object, I've usually either seen a mixin as a pure object representing a partial service schema (such as shown in the [docs example](https://moleculer.services/docs/0.14/services.html#Mixins)) or an invoked function that returns an object representing a partial service schema.  Your class seems to represent a service that you want to reuse, but that's not really how mixins are intended to work AFAIK.

Reply:
@devalexandre I also noticed that you're using `Service.mergeSchemas` which is a static method which is intended to merge a mixin into the mixed service.  It gets called automatically to merge the declared mixins when calling `parseServiceSchema`.  Honestly, I'm not even sure that some of these static member should be expected to be used by external consumers.  I would anticipate that those should really be considered private members and not really intended for external use.  I see they have been added to the TS typing, but my instinct is that is a mistake.  @icebob What do you think there?

Reply:
@shawnmcknight  So the best way would be to use the mixin pattern, and remove the use of static methods and the use of partials?

https://github.com/moleculerjs/moleculer-template-project-typescript/blob/master/template/mixins/db.mixin.ts.

in the example given in the typescript template repository, this way also using ThisType and static method, I also think this is bad practice but as it is the default used I kept it that way, to avoid big changes

Reply:
> @icebob What do you think there?

yeah, the `merge...` static methods are only for internal usages.